lebedigital/openbis/expstep.py

```diff
@@ -0,0 +1,531 @@
+import logging
+import os
+import shutil
+from math import isnan
+from sys import exit
+
+import pandas as pd
+import yaml
+from termcolor import colored
+
+from lebedigital.openbis.interbis import Interbis
+
+
+class ExpStep:
+    def __init__(
+            self,
+            name: str = '',
+            type: str = '',
+            metadata: dict = None,
+            space: str = "",
+            project: str = "",
+            collection: str = "",
+            parents: list = None,
+            identifier: str = '',
+            permId: str = '',
+            sample_object = None,
+            datasets: list = None,
+            dataset_codes: list = None,
+    ) -> None:
+        """Creates an ExpStep Object as general python interface to the Experimental Step OpenBis object
+
+        Args:
+            name (str, optional): name of the experiment. Defaults to ''.
+            type (str, optional): type of the experiment, has to be a defined sample type in the Datastore. Defaults to ''.
+            metadata (dict, optional): metadata of your data, is defined by the sample type. Defaults to None.
+            space (str, optional): space in which the experiment should be saved. Defaults to "".
+            project (str, optional): project in which the experiment should be saved. Defaults to "".
+            collection (str, optional): collection in which the experiment should be saved. Defaults to "".
+            parents (list, optional): parents of the experiment. Defaults to None.
```

Comment:
You can define parents of a sample/experimental step in the openBIS datastore. They are there to show connection between two measurements/research steps/datasets/etc. For example we use it to have a mixture Sample which is a "parent" to the concrete measurement Sample because the concrete measurement was performed on concrete samples made using that mixture.